sdk/test/OpenTelemetry/TraceSpec.hs

```diff
@@ -208,6 +209,17 @@ spec = describe "Trace" $ do
         (lookupAttribute attrs "code.function") == Just (toAttribute @Text "g")
           && (lookupAttribute attrs "code.namespace") == Just (toAttribute @Text "OpenTelemetry.TraceSpec")
 
+    specify "Attribute length limit is respected" $ asIO $ do
+      p <- getGlobalTracerProvider
+      let t = makeTracer p "woo" tracerOptions
+      s <- createSpan t Context.empty "test_span" defaultSpanArguments
+      addAttribute s "attr1" ("looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong" :: Text)
+      addAttributes s (HM.singleton "attr2" (toAttribute @Text "looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong"))
+      s <- unsafeReadSpan s
+
+      lookupAttribute (spanAttributes s) "attr1" `shouldBe` Just (toAttribute @Text "looooooooooooooooooooooooooooooooooooooooooooooooo")
+      lookupAttribute (spanAttributes s) "attr2" `shouldBe` Just (toAttribute @Text "looooooooooooooooooooooooooooooooooooooooooooooooo")
```

Comment:
Mildly not obvious what this test is doing. Could you add comments explaining the input and expected output? I wonder if it would be clearer to state the expectation a bit more declaratively: the actual attribute has the prefix of the input and it's length is <= the limit.

Reply:
Yeah, I made this simpler.